x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/TransportEsqlQueryAction.java

```diff
@@ -321,12 +322,12 @@ private EsqlExecutionInfo createEsqlExecutionInfo(EsqlQueryRequest request) {
     private EsqlQueryResponse toResponse(Task task, EsqlQueryRequest request, Configuration configuration, Result result) {
         List<ColumnInfoImpl> columns = result.schema().stream().map(c -> {
             List<String> originalTypes;
-            if (c.originalTypes() == null) {
-                originalTypes = null;
-            } else {
+            if (c instanceof FieldAttribute fa && (originalTypes = fa.originalTypes()) != null) {
                 // Sort the original types so they are easier to test against and prettier.
-                originalTypes = new ArrayList<>(c.originalTypes());
+                originalTypes = new ArrayList<>(originalTypes);
```

Comment:
Prior version was a bit more readable with a second method call here.
Does `c.originalTypes()` compute attributes or simply return pre-computed list?

Reply:
Precomputed list.

Reply:
I think this makes it harder to read.